src/foam/nanos/auth/ChangePasswordView.js

```diff
@@ -9,267 +9,166 @@ foam.CLASS({
       }
     },
     {
-      class: 'Password',
-      name: 'newPassword',
-      section: 'resetPasswordSection',
-      view: {
-        class: 'foam.u2.view.PasswordView',
-        passwordIcon: true
-      },
-      minLength: 6
+      class: 'String',
+      name: 'mode',
+      documentation: 'name of password model used for this view',
+      factory: function() {
+        if ( window.location.hash.includes('personal-settings') ||
+             window.location.hash.includes('set-personal') ) {
+          return 'update';
+        }
+        const token = new URLSearchParams(location.search).get('token');
+        return token ? 'reset' : 'send';
+      }
     },
     {
-      class: 'Password',
-      name: 'confirmationPassword',
-      label: 'Confirm Password',
-      section: 'resetPasswordSection',
-      view: {
-        class: 'foam.u2.view.PasswordView',
-        passwordIcon: true
-      },
-      validationPredicates: [
-        {
-          args: ['newPassword', 'confirmationPassword'],
-          predicateFactory: function(e) {
-            return e.EQ(
-              foam.nanos.auth.ChangePasswordView.NEW_PASSWORD,
-              foam.nanos.auth.ChangePasswordView.CONFIRMATION_PASSWORD);
-          },
-          errorString: 'Passwords do not match.'
+      name: 'passwordModel',
+      documentation: 'password model used for this view',
+      factory: function() {
+        if ( this.mode === 'update' ) {
+          return this.UpdatePassword.create();
+        } else if ( this.mode === 'reset' ) {
+          return this.ResetPassword.create();
         }
-      ]
+        return this.SendPassword.create();
+      }
     }
   ],
 
-  messages: [
-    { name: 'TITLE', message: 'Reset your password' },
-    { name: 'SUBTITLE', message: 'Create a new password for your account' },
-    { name: 'SUCCESS_MSG', message: 'Your password was successfully updated.' }
-  ],
-
   methods: [
     {
-      name: 'makeHorizontal',
-      code: function(value) {
+      name: 'resizeInputFields',
+      code: function(isHorizontal) {
         // evaluation of this gridColumns on a property does not handle an evaluation, only an int, thus function here managing this.
-        this.cls_.getAxiomByName('originalPassword').gridColumns = value ? ( this.token ? 0 : 4 ) : 12;
-        this.cls_.getAxiomByName('newPassword').gridColumns = value ? ( this.token ? 6 : 4 ) : 12;
-        this.cls_.getAxiomByName('confirmationPassword').gridColumns = value ? ( this.token ? 6 : 4 ) : 12;
+        if ( isHorizontal ) {
+          if ( this.mode === 'update' ) {
+            this.UpdatePassword.ORIGINAL_PASSWORD.gridColumns = 4;
+            this.UpdatePassword.NEW_PASSWORD.gridColumns = 4;
+            this.UpdatePassword.CONFIRMATION_PASSWORD.gridColumns = 4;
+          } else if ( this.mode === 'reset' ) {
+            this.ResetPassword.NEW_PASSWORD.gridColumns = 6;
+            this.ResetPassword.CONFIRMATION_PASSWORD.gridColumns = 6;
```

Comment:
I think you'll be able to use a ternary operation here if you have the password model. 
```javascript
passwordModel.getAxiomByName('newPassword').gridColumns = passwordModel.cls_.name == 'UpdatePassword' ?  4 : 6;
```
This might not be completely correct but I hope it conveys the idea.

Reply:
For scalability, wouldn't it be better to separate the implementation for each mode? Currently, we only need to consider two modes (reset and update) so tertiary operators work better, but later on if we decide to add more modes that use password fields, the expression might not be so clear at first sight as we will need to chain the expression. 

What do you think?

Reply:
So now that we have the separate models for passwords, the gridColumn logic should be on the models. 

this view really should not care what model it is being given. It should be able to handle the models as they are.